DesignQueueUsingStack.java

```diff
@@ -0,0 +1,78 @@
+// Time Complexity :
+// Average time complexity : O(1) , but worst time complexity : O(n) as we push all elements from instack to outstack
+
+// Space Complexity : O(n) as we are using two stacks
+
+// Did this code successfully run on Leetcode : Yes
+
+// Any problem you faced while coding this : None
+
+//Solution:
+import java.util.*;
+class DesingQueueUsingStack {
+
+    static Stack<Integer> in = new Stack<>();
+    static Stack<Integer> out = new Stack<>();
+
+
+    /** Push element x to the back of queue. */
+    public static void push(int x) {
+        in.push(x); // Push elements into inStack
+    }
+
+    /** Removes the element from in front of queue and returns that element. */
+    public int pop() {
+        /* if outStack is empty, push all elements of inStack into OutStack*/
+        if(out.isEmpty()){
+            while(!in.isEmpty()){
+                out.push(in.pop());
+            }
+
+        }
+        //Edge case to check if outstack is empty and we try to pop from it
+        if(out.isEmpty()){
+            System.out.println("Queue is empty");
```

Comment:
please try to avoid SOP while submitting GIT pull request

Reply:
Sure, Noted.